frontend/src/pages/Author.jsx

```diff
@@ -56,8 +68,169 @@ class Author extends Component {
+        }
+
+        else if (this.state.isFollowing && !this.state.isSelf) {
+            followbutton = <Button negative onClick={this.sendUnfollowRequest}><Icon name = "user times"/>Unfollow</Button>
+        }
+
+        return <div>{followbutton}</div>;
+    
+    }
+
+    getloggedinAuthorIDandHost() {
+        const cookieauthorid = Cookies.get("userID"),
+            storeauthorid = store.getState().loginReducers.userId,
+            cookiehost = cookieauthorid.split("author/")[0],
+            storehost = store.getState().loginReducers.hostName;
+        let authorID;
+        let host;
+        if (cookieauthorid !== null) {
+            authorID = cookieauthorid;
+        } else if (storeauthorid !== null) {
+            authorID = storeauthorid;
+        }
+        if (cookiehost !== null) {
+            host = cookiehost;
+        } else if (storehost !== null) {
+            host = storehost;
+        }
+        return [authorID, host]
+    }
+
+    getFollowStatus() {
+        var authorID;
+        authorID = this.getloggedinAuthorIDandHost()
+        let urlPath = "/api/followers/" + utils.getShortAuthorId(this.props.location.state.fullAuthorId),
+            requireAuth = true;
+        HTTPFetchUtil.getRequest(urlPath, requireAuth)
+            .then((httpResponse) => {
+                if (httpResponse.status === 200) {
+                    httpResponse.json().then((results) => {
+                        for (var i = 0; i < results.authors.length; i++) {
+                            if (results.authors[i].id === authorID[0]){
+                                this.setState({
+                                    isFollowing: true
+                                })
+                            }
+                        }
+                    })
+                }
+            })
+            .catch((error) => {
+                console.error(error);
+        });    
+        } 
+
+    sendUnfollowRequest() {
+        var authorID;
+        authorID = this.getloggedinAuthorIDandHost()
+        let urlPath = "/api/unfollow/"
+		let body = {
+			query: "unfollow",
+			author: {
+				id: authorID[0],
+				host: authorID[1],
+			},
+			friend:{
+				id: this.props.location.state.fullAuthorId,
+				host: this.props.location.state.host,
+			}
+		}
+		HTTPFetchUtil.sendPostRequest(urlPath, true, body)
+            .then((httpResponse) => {
+                if (httpResponse.status === 200) {
+                    httpResponse.json().then((results) => {
+                        this.setState({
+                            isFollowing: false
+                        })
+						toast(
+							{
+								type: 'success',
+								icon: 'user',
+								title: 'Unfollow successful!',
+								description: <p>You are now following {this.props.location.state.displayName}</p>
+							}
+						);
+
+                    })
+				}
+				else{
+					console.log(httpResponse)
```

Comment:
If you don't need this console log, would you be able to remove this?

Reply:
ok